WebService/v1/Models/DeviceModelApiModel/DeviceModelApiModel.cs

```diff
@@ -41,33 +56,75 @@ public class DeviceModelApiModel
         public IDictionary<string, string> Metadata => new Dictionary<string, string>
         {
             { "$type", "DeviceModel;" + v1.Version.NUMBER },
-            { "$uri", "/" + v1.Version.PATH + "/devicemodels/" + this.Id }
+            { "$uri", "/" + v1.Version.PATH + "/devicemodels/" + this.Id },
+            { "$created", this.created.ToString(DATE_FORMAT) },
+            { "$modified", this.modified.ToString(DATE_FORMAT) }
         };
 
         public DeviceModelApiModel()
         {
+            this.ETag = string.Empty;
             this.Id = string.Empty;
             this.Version = string.Empty;
             this.Name = string.Empty;
             this.Description = string.Empty;
             this.Protocol = string.Empty;
+            this.Type = string.Empty;
             this.Simulation = new DeviceModelSimulation();
             this.Properties = new Dictionary<string, object>();
             this.Telemetry = new List<DeviceModelTelemetry>();
             this.CloudToDeviceMethods = new Dictionary<string, DeviceModelSimulationScript>();
         }
 
+        // Map API model to service model
+        public DeviceModel ToServiceModel(string id = "")
+        {
+            this.Id = id;
+
+            var result = new DeviceModel
+            {
+                ETag = this.ETag,
+                Id = this.Id,
+                Version = this.Version,
+                Name = this.Name,
+                Description = this.Description,
+                Type = this.Type,
+                Protocol = (IoTHubProtocol)Enum.Parse(typeof(IoTHubProtocol), this.Protocol, true),
```

Comment:
Enum.Parse will throw an exception if the value is not valid. Extend `ValidateInputRequest` to validate it (+tests) so this won't happen

Reply:
Done.